addons/sale/wizard/sale_order_discount.py

```diff
@@ -99,54 +99,55 @@ def _create_discount_lines(self):
         discount_product = self._get_discount_product()
 
         if self.discount_type == 'amount':
+            if not self.sale_order_id.amount_total:
+                return
+            discount_percentage = self.discount_amount / self.sale_order_id.amount_total
+        else: # so_discount
+            discount_percentage = self.discount_percentage
+        total_price_per_tax_groups = defaultdict(float)
+        for line in self.sale_order_id.order_line:
+            if not line.product_uom_qty or not line.price_unit:
+                continue
+            discounted_price = line.price_unit * (1 - (line.discount or 0.0)/100)
+            total_price_per_tax_groups[line.tax_id] += (discounted_price * line.product_uom_qty)
```

Comment:
About fixed taxes, maybe we could start by just removing all of them when computing the discounts and say it's a limitation? :roll_eyes:

Reply:
We should probably do that everywhere though, so I'd suggest doing it in another PR dedicated to it, as the problem appears at different places. :thinking: